### ISSUE TITLE
Stripe script - don't rely on window.onload to initialise

### ISSUE THREAD

**Issue by croxton** [solspace/craft3-freeform#1281]
### What happened?

I'm setting up a payment form that is swapped with ajax into the page (via htmx / Sprig), and have the `Freeform Script Insertion Location` set to `Inside form`.

Htmx will execute any `<script>` tags in content swapped into the dom which successfully initialises the main freeform.js plugin, however the separate stripe.js script relies on `window.onload()` to fire in order to set up payment fields and also create a MutationObserver to watch for any subsequent forms loaded into the dom. It looks to me like that observer doesn't actually need to be wrapped inside the `window.onload`?

https://github.com/solspace/craft-freeform/blob/4145cc20a7cdf40c192b1a3eccf41822d788e05f/packages/scripts/src/components/front-end/payments/stripe/elements.ts#L28

### How can we reproduce this?

`/my-freeform` routes to template that generates a html fragment - a standard payment form setup with Stripe:

```html
<div id="main">
   {% set form = craft.freeform.form('paymentForm') %}
   {{ form.render() }}
</div>
```

Somewhere in another template representing a full HTML page, we have link that loads the payment form html fragment into a placeholder :

```html
<script src="https://unpkg.com/htmx.org@1.9.12"></script>
<a href="/my-freeform" hx-target="#main" hx-select="#main" hx-swap="outerHTML" hx-push-url="true">Load freeform into div#main</a>

<div id="main"></div>
```

The form loads as expected but the Stripe payment field does not get initialised.

### Freeform Edition

Pro

### Freeform Version

5.2.0

### Craft Version

5.1.3

### When did this issue start?

Fresh install of Freeform

**Comment by croxton** [comment]
I solved it like this:

```js
 this.js([
          '/freeform/scripts/stripe.js',
          '/freeform/plugin.js'
      ]).then((init = false) => {
          if (init) {
              // trigger window.onload() manually for stripe.js
              // when the script is first loaded so it can initialise itself
              dispatchEvent(new Event('load'));
          }
          // setup form options
          this.setupForms();
      });
```        

However, it would be far better if there was a custom event I could trigger to init stripe.js, rather than the `window.load`, assuming one is even necessary.

**Comment by gustavs-gutmanis** [comment]
Hi @croxton,

Thanks for bringing this issue to light. You're absolutely right, it does not make sense to have it trigger on `window.onload()`. We will address this, and come back with a better solution in this thread eventually.

**Comment by kjmartens** [comment]
This should now be resolved in Freeform 5.2.2. 🙂

**Comment by croxton** [comment]
For anyone finding this in the future, as of Freeform 5.2.2 you can now initialise Stripe payments on forms loaded via ajax into your page using the custom `freeform-stripe-load` event:

```js
document.dispatchEvent(new CustomEvent('freeform-stripe-load'));
```
